### PR TITLE
Adding delay to re-run

### PR DESCRIPTION
**Pull request by bigmstone**
This commit adds necessary changes to st2client and st2api to accommodate delay
on re-run of actions.

### CODE REVIEW

File: st2api/st2api/controllers/v1/actionexecutions.py [hunk 1]
```diff
@@ -457,7 +463,7 @@ class ActionExecutionsController(BaseResourceIsolationControllerMixin,
                                  ActionExecutionsControllerMixin, ResourceController):
     """
         Implements the RESTful web endpoint that handles
-        the lifecycle of ActionExecutions in the system.
+        the lifecycle of ActionExecutios in the system.
```
Comment (m4dcoder) [inline st2api/st2api/controllers/v1/actionexecutions.py]: typo?

File: st2api/st2api/controllers/v1/actionexecutions.py [hunk 2]
```diff
@@ -406,6 +406,11 @@ def post(self, spec_api, id, requester_user, no_merge=False, show_secrets=False)
         if list(set(spec_api.reset) - set(spec_api.tasks)):
             raise ValueError('List of tasks to reset does not match the tasks to rerun.')
 
+        delay = None
+
+        if hasattr(spec_api, "delay") and isinstance(spec_api.delay, int):
```
Comment (Kami) [inline st2api/st2api/controllers/v1/actionexecutions.py]: I would just do ``delay = getattr(spec_api, 'delay', None)``, but that also works.

File: st2common/st2common/models/db/execution.py [hunk 3]
```diff
@@ -179,7 +179,7 @@ class ActionExecutionOutputDB(stormbase.StormFoundationDB):
     runner_ref = me.StringField(required=True)
     timestamp = ComplexDateTimeField(required=True, default=date_utils.get_datetime_utc_now)
     output_type = me.StringField(required=True, default='output')
-    delay = me.IntField(required=True, default=0)
+    delay = me.IntField()
```
Comment (Kami) [inline st2common/st2common/models/db/execution.py]: Any particular reason why you removed that? I guess required with a default value doesn't make much sense / difference.

Comment (bigmstone) [reply st2common/st2common/models/db/execution.py]: Making it consistent with other delay references in other collections. No other reason.

**Pull request merged**